### ISSUE TITLE
Base class InjectionMethod not populated

### ISSUE THREAD

**Issue by mgth** [z4kn4fein/stashbox#103]
Dependency properties in base class are OK, but method is not injected.
by the way, I did not find a way to inject methods without using attributs (or declaring my own attribut). Is there one?

```csharp
   public class A
    { }
    public class B
    { }

    public class BaseClass
    {
        [Dependency]
        public A A { get; set; }
        public bool DoneA { get; set; }
        
        [InjectionMethod]
        public void InjectA()
        {
            DoneA = true;
        }
    }

    public class MainClass : BaseClass
    {
        [Dependency]
        public B B { get; set; }
        public bool DoneB { get; set; }

        [InjectionMethod]
        public void InjectB()
        {
            DoneB = true;
        }
    }

    public class BaseClassMethod
    {
        [Fact]
        public void Test()
        {
            var container = new StashboxContainer();
            container.Register<A>().Register<B>().Register<MainClass>();

            var main = container.Resolve<MainClass>();

            Assert.IsType<B>(main.B);
            Assert.IsType<A>(main.A);

            Assert.True(main.DoneA);
            Assert.True(main.DoneB);
        }
    }   
```

**Comment by z4kn4fein** [comment]
Thanks for reporting and for the PR!
Yes, there is an additional method to achieve something similar, you can configure an initializer delegate on registration:
`.Register<MainClass>(c => c.WithInitializer(mainClass => mainClass.InjectB()));`

To resolve dependencies, you can use a delegate with an extra `IDependencyResolver` parameter like:
`.Register<MainClass>(c => c.WithInitializer((mainClass, resolver) => mainClass.InjectB(resolver.Resolve<SomeDependency>())));`

**Comment by z4kn4fein** [comment]
Released in [v3.6.3](https://github.com/z4kn4fein/stashbox/releases/tag/3.6.3)
